### PR TITLE
In 148382187 refactor legacy fields

### PR DESCRIPTION
**Pull request by saraOSI**
Pull participant controller actions into service objects to clean up an overly long controller.

### CODE REVIEW

File: spec/services/update_participant_spec.rb [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        end
+      end
+
+      context 'when one of the phone numbers is new, and one existed previously' do
+        before do
+          participant.phone_numbers = [phone_number1]
+          participant.save!
+        end
+
+        it 'only creates one new address' do
```
Comment (ThomasOSI) [inline spec/services/update_participant_spec.rb]: phone number

File: spec/services/update_participant_spec.rb [hunk 2]
```diff
@@ -0,0 +1,136 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe UpdateParticipant do
+  describe '#call' do
```
Comment (ThomasOSI) [inline spec/services/update_participant_spec.rb]: Do we need tests for removing addresses & phone numbers?

File: spec/factories/legacy_descriptor_factory.rb [hunk 3]
```diff
@@ -2,11 +2,14 @@
...
-    legacy_ui_id '0647-2921-1069-4001420'
-    legacy_last_updated '2002-08-28T14:29:59.000-0700'
-    legacy_table_name 'CLIENT_T'
-    legacy_table_description 'Client'
+    legacy_id { FFaker::Guid.guid }
+    legacy_ui_id { FFaker::Guid.guid }
+    legacy_last_updated { FFaker::Time.datetime }
+    legacy_table_name { %w[CLIENT_T VIC_T PERP_T REPORT_T].sample }
+    legacy_table_description { %w[Client Victim Perpetrator Reporter].sample }
+  end
```
Comment (ThomasOSI) [inline spec/factories/legacy_descriptor_factory.rb]: :thumbsup:

File: spec/services/update_participant_spec.rb (hunk 2 above)
Comment (saraOSI) [inline spec/services/update_participant_spec.rb]: I think we have request specs that cover this, but adding additional tests here is a good idea.

File: app/services/update_participant.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+# Service to update a participant given a set of params
+class UpdateParticipant
+  def initialize; end
+
+  def call(participant:, params:, address_params:, phone_params:)
```
Comment (jgedeon-osi) [inline app/services/update_participant.rb]: We never need to update legacy_descriptor?

Comment (saraOSI) [inline app/services/update_participant.rb]: Not currently. We should be easily able to add that functionality if/when we need to.

File: spec/services/update_participant_spec.rb (hunk 2 above)
Comment (thomasramirez) [inline spec/services/update_participant_spec.rb]: Thanks for adding.  These look good.

**Pull request merged**